doc/csharp/csharp-null-ref-types-proposal.md

```diff
@@ -0,0 +1,170 @@
+reference types in the generated code.
+
+## Background
+
+*Nullable reference types* refers to a group of optional features introduced 
+in C# 8.0 to enable the compiler to do static flow analysis to determine if a
+variable might be null before it is dereferenced. They are compile time features
+that do not affect the runtime.
+
+Various issues have been raised asking for the code generated by the protocol
+buffers compiler (`protoc`) and the gRPC C# compiler plugin 
+(`grpc_csharp_plugin`) to support nullable reference types, including:
+
+- [Investigate support for nullable reference types in generated code](https://github.com/grpc/grpc-dotnet/issues/606)
+- [Add option to generate C# nullable annotations](https://github.com/protocolbuffers/protobuf/issues/6632)
+- [C#: Consider exposing protobuf optional fields as nullable properties](https://github.com/protocolbuffers/protobuf/issues/9083)
+- [Support C# 8 nullable type references in Grpc.Tools](https://github.com/grpc/grpc/issues/20729)
+
+## Proposal
+
+The proposal is to maintain the status quo and to keep code generation 
+unchanged. Reasons are given below.
+
+## Rationale
+
+The reasons for currently maintaining the status quo are summarised here:
+1. The generated code is not in a nullable context
+2. *.NET Framework* projects default to using C# 7.3
+3. Building gRPC projects with nullable enabled works today
+4. Breaking backwards compatibility with gRPC service classes
+
+There are very few disadvantages to not supporting nullable reference types in
+the generated code. These are given later.
+
+The above reasons are explained in more detail:
+
+### 1. The generated code is not in a nullable context
+
+The generated code is not in a nullable context. The `// <auto-generated>` 
+comment in the code disables it. No nullable checks are made and no compiler 
+warnings are generated for this code, even if the code is compiled in a project 
+with nullable checks enabled.
+
+### 2. .NET Framework projects default to using C# 7.3
+
+*.NET Framework* projects default to using compiler version C# 7.3 which does
+not support nullable reference type features.
+
+*Visual Studio 2022* does not let you change to a later compiler version for
+*.NET Framework* projects:
+- In *Advanced Build Settings | Language version* the drop down to select the 
+language version is greyed out and says “*Automatically selected based on 
+framework version*”
+
+See https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version 
+
+Note: It is possible to edit the project file and explicitly set the language 
+version to a later version, but other language features that require runtime 
+changes in the CLR will not work. C# 7.3 is the latest language version that is 
+fully supported with *.NET Framework* projects.
+
+### 3. Building gRPC projects with nullable enabled works today
+
+There does not appear to be a problem building *.NET Core* projects that have
```

Comment:
I don't know what the preferred way of saying ".NET Core and .NET" is these days. Possibly just that. But given that all .NET Core platforms are now out of MS support (IIRC) - I think it would at least be better to drop the Core. Maybe we need an introductory paragraph with terminology - I'd be fine with "Within this document, references to '.NET Core' are intended to include .NET 6.0+" or similar.